src/osp/Resource/AssetImporter.cpp

```diff
@@ -67,20 +67,23 @@ void osp::AssetImporter::load_sturdy_file(std::string const& filepath, Package&
         std::cout << "Error: couldn't open file: " << filepath << "\n";
     }
 
-    load_sturdy(gltfImporter, pkg);
+    // repurpose filepath into unique identifier for file resources
+    std::string dataResourceName = filepath.substr(0, filepath.find('.')) + ':';
```

Comment:
This is expensive.

substr is an allocate + copy all by itself, and then you're appending data to it, which is another allocate and copy.

Probably this should use the string_concat function, and that substr call should be against a string_view to avoid paying for an unnecessary allocate and copy.